src/VisualStudio/Core/Impl/Options/GridOptionPreviewControl.xaml.cs

```diff
@@ -93,5 +106,264 @@ internal override void Close()
                 this.ViewModel.Dispose();
             }
         }
+
+        internal void Generate_Save_Editorconfig(object sender, System.Windows.RoutedEventArgs e)
+        {
+            var optionSet = this.ViewModel.ApplyChangedOptions(this.OptionService.GetOptions());
+            var editorconfig = new StringBuilder();
+            Generate_Editorconfig(optionSet, _language, editorconfig, _getLangaugeSpecificEditorConfigOptions);
+            var sfd = new System.Windows.Forms.SaveFileDialog
+            {
+                Filter = "All files (*.*)|",
+                FileName = ".editorconfig",
+                Title = "Save .editorconfig file"
+            };
+            if (sfd.ShowDialog() == System.Windows.Forms.DialogResult.OK)
+            {
+                IOUtilities.PerformIO(() =>
+                {
+                    var path = sfd.FileName;
+                    using (var sw = new StreamWriter(File.Create(path)))
+                    {
+                        sw.Write(editorconfig);
+                        sw.Close();
+                    }
+                });
+            }
+        }
+
+        internal static void Generate_Editorconfig(
+            OptionSet optionSet,
+            string language,
+            StringBuilder editorconfig,
+            Action<OptionSet, StringBuilder> getLangaugeSpecificEditorConfigOptions)
+        {
+            GenerateEditorconfig_CoreSettings(optionSet, language, editorconfig);
+            GenerateEditorconfig_DotNetSettings(optionSet, language, editorconfig);
+
+            getLangaugeSpecificEditorConfigOptions(optionSet, editorconfig);
+        }
+
+        private static void GenerateEditorconfig_CoreSettings(OptionSet optionSet, string language, StringBuilder editorconfig)
```

Comment:
nit: consider putting all this logic in a separate place, not in this codebehind